### PR TITLE
Sync june k8s patches to dev-v2.7 branch 

### PR DESCRIPTION
**Pull request by kinarashah**
- `dev-v2.7-2023-06-patches` for June k8s patch versions releasing sometime in July. (`rke/v1.4` branch) 
- `dev-v2.7` for July k8s patch versions and other fixes with Rancher Q3-2.7.x milestone. Will have all changes from above branch and more. (`rke/v1.4.8` branch)

### CODE REVIEW

File: pkg/rke/k8s_rke_system_images.go [hunk 1]
```diff
@@ -9034,7 +9078,51 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			WindowsPodInfraContainer:  "rancher/mirrored-pause:3.7",
 			Nodelocal:                 "rancher/mirrored-k8s-dns-node-cache:1.22.10",
 		},
-		// Enabled in Rancher v2.7.4
+		// Enabled out of band post v2.7.5
+		"v1.25.11-rancher1-1": {
+			Etcd:                      "rancher/mirrored-coreos-etcd:v3.5.4",
```
Comment (jiaqiluo) [inline pkg/rke/k8s_rke_system_images.go]: `rancher/mirrored-coreos-etcd:v3.5.6` is used in `v1.25.9-rancher2-2`, any reason for going back to an older tag?

File: pkg/rke/k8s_rke_system_images.go [hunk 2]
```diff
@@ -8858,6 +8858,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
+			WeaveCNI:                  "weaveworks/weave-npc:2.8.1",
+			AciCniDeployContainer:     "noiro/cnideploy:5.2.3.6.1d150da",
+			AciHostContainer:          "noiro/aci-containers-host:5.2.3.6.1d150da",
+			AciOpflexContainer:        "noiro/opflex:5.2.3.6.1d150da",
+			AciMcastContainer:         "noiro/opflex:5.2.3.6.1d150da",
+			AciOpenvSwitchContainer:   "noiro/openvswitch:5.2.3.6.1d150da",
+			AciControllerContainer:    "noiro/aci-containers-controller:5.2.3.6.1d150da",
+			AciGbpServerContainer:     "noiro/gbp-server:5.2.3.6.1d150da",
+			AciOpflexServerContainer:  "noiro/opflex-server:5.2.3.6.1d150da",
+			PodInfraContainer:         "rancher/mirrored-pause:3.7",
```
Comment (jiaqiluo) [inline pkg/rke/k8s_rke_system_images.go]: All those tags are older than the ones used in `v1.24.13-rancher2-2`. 
It looks like a mistaken to me unless there is a reason I am unware of.

File: pkg/rke/k8s_rke_system_images.go [hunk 3]
```diff
@@ -9034,7 +9078,51 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
+			WeaveNode:                 "weaveworks/weave-kube:2.8.1",
+			WeaveCNI:                  "weaveworks/weave-npc:2.8.1",
+			AciCniDeployContainer:     "noiro/cnideploy:5.2.3.6.1d150da",
+			AciHostContainer:          "noiro/aci-containers-host:5.2.3.6.1d150da",
+			AciOpflexContainer:        "noiro/opflex:5.2.3.6.1d150da",
+			AciMcastContainer:         "noiro/opflex:5.2.3.6.1d150da",
+			AciOpenvSwitchContainer:   "noiro/openvswitch:5.2.3.6.1d150da",
+			AciControllerContainer:    "noiro/aci-containers-controller:5.2.3.6.1d150da",
+			AciGbpServerContainer:     "noiro/gbp-server:5.2.3.6.1d150da",
+			AciOpflexServerContainer:  "noiro/opflex-server:5.2.3.6.1d150da",
```
Comment (jiaqiluo) [inline pkg/rke/k8s_rke_system_images.go]: Same issue: older tags are used in the template of the new k8s version.

File: pkg/rke/k8s_rke_system_images.go [hunk 4]
```diff
@@ -9078,6 +9166,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
 			WindowsPodInfraContainer:  "rancher/mirrored-pause:3.7",
 			Nodelocal:                 "rancher/mirrored-k8s-dns-node-cache:1.22.20",
 		},
+		// Enabled out of band post v2.7.5
+		"v1.26.6-rancher1-1": {
+			Etcd:                      "rancher/mirrored-coreos-etcd:v3.5.4",
```
Comment (jiaqiluo) [inline pkg/rke/k8s_rke_system_images.go]: Same issue: "v1.26.4-rancher2-1" uses  "rancher/mirrored-coreos-etcd:v3.5.6"

File: pkg/rke/k8s_rke_system_images.go [hunk 5]
```diff
@@ -9078,6 +9166,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
+			WeaveNode:                 "weaveworks/weave-kube:2.8.1",
+			WeaveCNI:                  "weaveworks/weave-npc:2.8.1",
+			AciCniDeployContainer:     "noiro/cnideploy:5.2.3.6.1d150da",
+			AciHostContainer:          "noiro/aci-containers-host:5.2.3.6.1d150da",
+			AciOpflexContainer:        "noiro/opflex:5.2.3.6.1d150da",
+			AciMcastContainer:         "noiro/opflex:5.2.3.6.1d150da",
+			AciOpenvSwitchContainer:   "noiro/openvswitch:5.2.3.6.1d150da",
+			AciControllerContainer:    "noiro/aci-containers-controller:5.2.3.6.1d150da",
+			AciGbpServerContainer:     "noiro/gbp-server:5.2.3.6.1d150da",
+			AciOpflexServerContainer:  "noiro/opflex-server:5.2.3.6.1d150da",
```
Comment (jiaqiluo) [inline pkg/rke/k8s_rke_system_images.go]: Same issue as the above.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
We need to fix multiple tags in the new template that are older than the ones in the previously released templates.

**Comment by kinarashah** [comment]
@jiaqiluo Thanks for calling it out. ~Looks like since June patches were introduced in custom branch before v2.7.5 Kubernetes versions were introduced, they were copied from older templates than the latest patch versions.~ @doflamingo721 Could you fix this in `dev-v2.7-2023-06-patches` so we don't downgrade versions with the new OOB release? 

EDIT: correction, the custom branch also had these versions already so it wasn't because of that. I missed these during review, @doflamingo721 two things to clarify: 
- We discussed etcd downgrade and decided it needs to be at 3.5.4, but only for v1.24.x because v1.24.x is at 3.5.4. Not other versions, as they can continue to be at 3.5.6. 
- ACI was updated to a new version in the latest released versions, so they need to be updated in these versions as well.

File: pkg/rke/k8s_rke_system_images.go (hunk 1 above)
Comment (doflamingo721) [reply pkg/rke/k8s_rke_system_images.go]: There was a misunderstanding while updating the June patches. The version was supposed to be downgraded for 1.24. I have now raised a PR to rectify this in the June patches custom branch.

**Comment by doflamingo721** [comment]
@jiaqiluo @kinarashah There was some misunderstanding while updating the June patches. I have rectified this in a new PR I have created [here](https://github.com/rancher/kontainer-driver-metadata/pull/1171).

**Review by jiaqiluo — APPROVED** [review]
LGTM

**Pull request merged**